mantidproject/mantid
# Indirect - Corrections Interfaces - Add Plot Current Preview

Currently in the Indirect Corrections interface, the ContainerSubtraction and ApplyPaalmanPings tabs do not have the functionality for plotting the current preview (mini-plot).

Added functionality for plotting current preview in the ContainerSubtraction and ApplyPaalmanPings interfaces.

**To test:**
1. Navigate to Interfaces > Indirect > Corrections.
2. To test each of the interfaces:
    1. In the ContainerSubtraction interface/tab:
        1. Enter a reduced file (*_red.nxs) or workspace (*_red) into the 'Sample' field.
        2. Once the graph in the preview (mini-plot) appears, click 'Plot Current Preview'.
        3. Ensure the plot matches the preview and then close the plot.
        4. Enter a reduced container file (*_red.nxs) into the 'Container' field.
        5. Once the graph in the preview (mini-plot) updates, click 'Plot Current Preview'.
        6. Ensure the plot matches the preview and then close the plot.
        7. Change the number in the spinner of the 'Spectrum' field.
        8. Once the graph in the preview (mini-plot) appears, click 'Plot Current Preview'.
        9. Ensure the plot matches the preview and then close the plot.
        10. Run the algorithm.
        11. Once the algorithm has completed and the preview has updated, click 'Plot Current Preview'.
        12. Ensure the plot matches the preview.
    2. In the ApplyPaalmanPings interface/tab:
        1. Enter a reduced file (*_red.nxs) or workspace (*_red) into the Input field.
        2. Once the graph in the preview (mini-plot) appears, click 'Plot Current Preview'.
        3. Ensure the plot matches the preview and then close the plot.
        4. Check the checkbox labelled 'Use Container' and enter a reduced container file/workspace into the field.
        5. Once the graph in the preview (mini-plot) updates, click 'Plot Current Preview'.
        6. Ensure the plot matches the preview and then close the plot.
        7. Change the number in the spinner of the 'Spectrum' field.
        8. Once the graph in the preview (mini-plot) updates, click 'Plot Current Preview'.
        9. Ensure the plot matches the preview and then close the plot.
        10. Run the algorithm.
        11. Once the algorithm has completed and the preview has updated, click 'Plot Current Preview'.
        12. Ensure the plot matches the preview.


Fixes #19094. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
